pkg/providers/instance/suite_test.go

```diff
@@ -107,24 +108,34 @@ var _ = Describe("InstanceProvider", func() {
 				},
 			},
 		})
-	})
 
-	It("should return an ICE error when all attempted instance types return an ICE error", func() {
-		ExpectApplied(ctx, env.Client, nodeClaim, nodePool, nodeClass)
-		zones := []string{"1", "2", "3"}
-		for _, zone := range zones {
-			azureEnv.UnavailableOfferingsCache.MarkUnavailable(ctx, "SubscriptionQuotaReached", "Standard_D2_v2", fmt.Sprintf("%s-%s", fake.Region, zone), v1alpha5.CapacityTypeSpot)
-			azureEnv.UnavailableOfferingsCache.MarkUnavailable(ctx, "SubscriptionQuotaReached", "Standard_D2_v2", fmt.Sprintf("%s-%s", fake.Region, zone), v1alpha5.CapacityTypeOnDemand)
-		}
-		instanceTypes, err := cloudProvider.GetInstanceTypes(ctx, nodePool)
-		Expect(err).ToNot(HaveOccurred())
-
-		// Filter down to a single instance type
-		instanceTypes = lo.Filter(instanceTypes, func(i *corecloudprovider.InstanceType, _ int) bool { return i.Name == "Standard_D2_v2" })
-
-		// Since all the offerings are unavailable, this should return back an ICE error
-		instance, err := azureEnv.InstanceProvider.Create(ctx, nodeClass, nodeClaim, instanceTypes)
-		Expect(corecloudprovider.IsInsufficientCapacityError(err)).To(BeTrue())
-		Expect(instance).To(BeNil())
+		azureEnv.Reset()
+		azureEnvNonZonal.Reset()
 	})
+
+	var ZonalAndNonZonalRegions = []TableEntry{
+		Entry("zonal", azureEnv, cloudProvider),
+		Entry("non-zonal", azureEnvNonZonal, cloudProviderNonZonal),
+	}
+
+	DescribeTable("should return an ICE error when all attempted instance types return an ICE error",
```

Comment:
Do we plan to make more of our tests table-driven?

Reply:
No specific plans, but we can definitely use this pattern where it makes sense.